### PR TITLE
Send data from sessionStorage via email. Related #70.

### PR DESCRIPTION
**Pull request by RhodesPeter**
- Collect data from sessionStorage and send to back-end
- Create email-body.js module to construct the email string
- Send the email 

This is what the completed email looks like this:
screen shot 2017-03-31 at 23 06 42

### CODE REVIEW

File: public/index.js [hunk 1]
```diff
@@ -175,6 +175,7 @@
     var emailAddress = emailRecipient.value;
     var http = new XMLHttpRequest();
     http.open("POST", '/finished', true);
-    http.send(emailAddress);
+    var payload = JSON.stringify([emailAddress, sessionStorage]);
```
Comment (des-des) [inline public/index.js]: Sorry to have missed this, but you need to make set a header (`application/json`) on this request.
Also, this should not be an array. It should be an object: 
```js
  {  emailAddress: emailAddress, emailData: sessionStorage }
```

File: src/email-body.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
+const bodyStyle = `background: #60ADD6;
```
Comment (des-des) [inline src/email-body.js]: You have a html template engine setup on your server (handlbars). Use it.

File: src/routes.js [hunk 3]
```diff
@@ -130,7 +130,9 @@ module.exports = [
     handler: (request, reply) => {
       reply({status: 'ok'});
       if (process.env.NODE_ENV === 'testing'){ return; }
-      sendMail(request.payload, function(err, info){
+      var payload = JSON.parse(request.payload);
+      var emailBody = JSON.stringify(payload[1]);
```
Comment (des-des) [inline src/routes.js]: See my comment on the xhr request.
You want:
```js
const { emalAddress, emailBody } = payload
```
at the top of this function. (you dont have to use es6 though )

File: src/email-body.js (hunk 2 above)
Comment (RhodesPeter) [inline src/email-body.js]: Hi @des-des - How should I go about this? Please could you give me some more guidance. Is it just the CSS or should I be creating this HTML string in the front-end with handlebars?

**Comment by RhodesPeter** [comment]
Hi @des-des - I have pushed up a new version. The email is being sent with the right information and it now comes through handlebars rather than in the backend as you requested, however we can't give our email.hbs access to our main.css file. Please could you point me in the right direction? Thank you :)

File: src/email-body.js (hunk 2 above)
Comment (des-des) [inline src/email-body.js]: @RhodesPeter have you asked @skibinska? We have walked through how to use handlebars server side without using the reply interface

**Comment by des-des** [comment]
@RhodesPeter you can inline the css (just as you were doing in your original version), also, your hbs is crying out for a loop

something like
```hbs
{{#each emailParagraph}}
<h2 style=" margin: 0 0 23px 0">{{ title }}</h2>
<p style="margin: 0;">{{text}}</p>
{{#endeach}}
```

**Comment by RhodesPeter** [comment]
Thanks @des-des.

**Comment by RhodesPeter** [comment]
Hi @des-des, I have added the css inline. 

We spent some time creating the each loop with Handlebars but in the end is was creating much more code in our app elsewhere. The main reason this was happening is because sessionStorage orders things alphabetically and using the each loop meant we lost control of which order the values were being displayed in the email.

**Pull request merged**